code/modules/mob/living/carbon/xenomorph/castes/warrior/warrior.dm

```diff
@@ -13,112 +13,56 @@
 	tier = XENO_TIER_TWO
 	upgrade = XENO_UPGRADE_NORMAL
 	bubble_icon = "alienroyal"
-	///How many stacks of combo do we have ? Interacts with every ability.
-	var/combo = 0
-	///Abilities with empowered interactions
-	var/list/empowerable_actions = list(
-		/datum/action/ability/activable/xeno/fling,
-		/datum/action/ability/activable/xeno/toss,
-		/datum/action/ability/activable/xeno/punch,
-		/datum/action/ability/activable/xeno/punch/jab,
-	)
+
+
 // ***************************************
 // *********** Icons
 // ***************************************
 /mob/living/carbon/xenomorph/warrior/handle_special_state()
-	if(agility)
+	var/datum/action/ability/xeno_action/toggle_agility/agility_action = actions_by_path[/datum/action/ability/xeno_action/toggle_agility]
+	if(agility_action?.ability_active)
 		icon_state = "Warrior Agility"
 		return TRUE
 	return FALSE
 
 /mob/living/carbon/xenomorph/warrior/handle_special_wound_states(severity)
 	. = ..()
-	if(agility)
+	var/datum/action/ability/xeno_action/toggle_agility/agility_action = actions_by_path[/datum/action/ability/xeno_action/toggle_agility]
+	if(agility_action?.ability_active)
 		return "warrior_wounded_agility_[severity]"
 
+
 // ***************************************
 // *********** Mob overrides
 // ***************************************
 /mob/living/carbon/xenomorph/warrior/stop_pulling()
 	if(isliving(pulling) && !isxeno(pulling))
-		var/mob/living/L = pulling
+		var/mob/living/living_target = pulling
 		grab_resist_level = 0 //zero it out
-		L.SetStun(0)
-		UnregisterSignal(L, COMSIG_LIVING_DO_RESIST)
+		living_target.SetStun(0)
+		UnregisterSignal(living_target, COMSIG_LIVING_DO_RESIST)
 	..()
 
 /mob/living/carbon/xenomorph/warrior/start_pulling(atom/movable/AM, force = move_force, suppress_message = TRUE, lunge = FALSE)
 	if(!check_state())
 		return FALSE
-
-	if(agility)
-		balloon_alert(src, "Cannot in agility mode")
-		return FALSE
-
-	var/mob/living/L = AM
-
-	if(isxeno(L))
-		var/mob/living/carbon/xenomorph/X = L
-		if(issamexenohive(X)) //No neckgrabbing of allies.
-			return ..()
-
+	var/mob/living/living_target = AM
+	if(isxeno(living_target) && issamexenohive(living_target))
+		return ..()
 	if(lunge && ..())
-		return neck_grab(L)
-
-	. = ..(L, force, suppress_message)
+		return neck_grab(living_target)
+	. = ..(living_target, force, suppress_message)
 
-/mob/living/carbon/xenomorph/warrior/proc/neck_grab(mob/living/L)
+/mob/living/carbon/xenomorph/warrior/proc/neck_grab(mob/living/living_target)
 	GLOB.round_statistics.warrior_grabs++
 	SSblackbox.record_feedback("tally", "round_statistics", 1, "warrior_grabs")
 	setGrabState(GRAB_NECK)
-	ENABLE_BITFIELD(L.restrained_flags, RESTRAINED_NECKGRAB)
-	RegisterSignal(L, COMSIG_LIVING_DO_RESIST, TYPE_PROC_REF(/atom/movable, resisted_against))
-	L.drop_all_held_items()
-	L.Paralyze(0.1 SECONDS)
-	visible_message(span_xenowarning("\The [src] grabs [L] by the throat!"), \
```

Comment:
I've been removing chat messages on most things I've done because Tivi wanted it this way (at least, a long time ago) in favor of balloon alerts. I just put one there regardless,